src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1620,7 +1620,7 @@ static void watch_connectivity_state_locked(grpc_exec_ctx *exec_ctx, void *arg,
     external_connectivity_watcher_list_append(w->chand, w);
     GRPC_CLOSURE_RUN(exec_ctx, w->watcher_timer_init, GRPC_ERROR_NONE);
     GRPC_CLOSURE_INIT(&w->my_closure, on_external_watch_complete, w,
```

Comment:
If `on_external_watch_complete()` is now going to run in the client channel combiner, please rename it to `on_external_watch_complete_locked()`.

Reply:
Fixed in latest update